axopy/timing.py

```diff
@@ -78,6 +78,89 @@ def reset(self):
         self.count = 0
 
 
+class StepCounter(Counter):
+    """Multiple step counter.
+       Counts to a given number then transmits a timeout event.
+       Transmits events at multiple values up to timeout.
+
+    Parameters
+    ----------
+    max_count : int
+        Number of iterations to go through before transmitting the `timeout`
+        event. Must be greater than 1.
+    reset_on_timeout : bool, optional
+        Specifies whether or not the timer should reset its count back to zero
+        once the timeout event occurs. The default behavior is to reset.
+
+    Attributes
+    ----------
+    count : int
+        Current count.
+    timeout : Transmitter
+        Transmitted when ``max_count`` has been reached.
+
+    Examples
+    --------
+    Basic usage:
+
+    >>> from axopy.timing import StepCounter
+    >>> timer = StepCounter(3)
+    >>> timer.add_step(1, function1)
+    >>> timer.add_step(2, function2)
+    >>> timer.increment()
+    "function 1"
+    >>> timer.increment()
+    "function 2"
+    >>> timer.increment()
+    >>> timer.count
+    0
+    """
+
+    timeout = Transmitter()
+
+    # README: No way to create an array of pyqtSignal(s) ...
```

Comment:
No I understand what the problem is.. This is strange indeed, I tried to use a dict as a workaround but that didn't work either. I guess this issue originates from the fact that PyQt is nothing but Python binders for C++ code...